pandas/tests/indexes/datetimes/test_tools.py

```diff
@@ -114,6 +114,17 @@ def test_to_datetime_format_integer(self, cache):
         result = to_datetime(s, format='%Y%m', cache=cache)
         assert_series_equal(result, expected)
 
+    @pytest.mark.parametrize('int_date, expected', [
+        [20121030, datetime(2012, 10, 30)],
+        [20129930, 20129930],
+        [20120011, 20120011],
+        [2012010101, 2012010101]])
+    def test_int_to_datetime_format_YYYYMMDD_typeerror(self, int_date,
```

Comment:
can you try with a too short as well?

Reply:
Hi @jreback,
Thanks for feedback. 
I reworded whatsnew message and added a couple of short dates. 

Also I found result can be unpredictable for some short dates (either in integer or string format) with length > 4.  
```python
pd.to_datetime(12912, format="%Y%m%d", errors='ignore')
Out[4]: 12912
pd.to_datetime(21213, format="%Y%m%d", errors='ignore')
Out[5]: datetime.datetime(2, 12, 13, 0, 0)
```
Nevertheless it's not connected to this TypeError.
I'll add these examples to corresponding issue.

Reply:
that is weird, these should for sure fail, but maybe going down an odd path. ok can address in another issue.

Reply:
I added it to #14487. It's all about how format '%Y%m%d is treated.